### PR TITLE
Migrate spec lib to spector

### PR DESCRIPTION
**Pull request by timotheeguerin**
Combine the spec-lib package into spector as this is a bit too much extra packages

### CODE REVIEW

File: packages/spector/src/scenarios-resolver.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { Operation } from "@typespec/compiler";
 import { isSharedRoute } from "@typespec/http";
 import { ScenarioMockApi } from "@typespec/spec-api";
-import { Scenario } from "@typespec/spec-lib";
 import { dirname, join, relative, resolve } from "path";
 import pc from "picocolors";
 import { pathToFileURL } from "url";
+import { Scenario } from "./lib/decorators.js";
```
Comment (timotheeguerin) [inline packages/spector/src/scenarios-resolver.ts:7]:
```suggestion
import type { Scenario } from "./lib/decorators.js";
```

File: packages/spector/src/lib/tsp-index.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import type { TypeSpecSpecLibDecorators } from "../generated-defs/TypeSpec.SpecLib.js";
+import { TypeSpecSpecLibDecorators } from "../../generated-defs/TypeSpec.SpecLib.js";
```
Comment (timotheeguerin) [inline packages/spector/src/lib/tsp-index.ts:1]:
```suggestion
import type { TypeSpecSpecLibDecorators } from "../../generated-defs/TypeSpec.SpecLib.js";
```

**Review by chrisradek — APPROVED** [review]

File: packages/http-specs/specs/authentication/api-key/main.tsp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import "@typespec/http";
-import "@typespec/spec-lib";
+import "@typespec/spector";
 
 using Http;
 using SpecLib;
```
Comment (sarangan12) [inline packages/http-specs/specs/authentication/api-key/main.tsp:5]: @timotheeguerin Do we want to keep this as using SpecLib? Or using Spector?

Comment (timotheeguerin) [reply packages/http-specs/specs/authentication/api-key/main.tsp:5]: ah yeah good point will change too

File: packages/spector/generated-defs/TypeSpec.SpecLib.ts-test.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /** An error here would mean that the decorator is not exported or doesn't have the right name. */
-import { $decorators } from "@typespec/spec-lib";
+import { $decorators } from "@typespec/spector";
 import type { TypeSpecSpecLibDecorators } from "./TypeSpec.SpecLib.js";
```
Comment (sarangan12) [inline packages/spector/generated-defs/TypeSpec.SpecLib.ts-test.ts:3]: Probably from `TypeSpecSpecLibDecorators` to `TypeSpecSpectorDecorators`

File: packages/spector/src/lib/decorators.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ import {
   ScenarioDecorator,
   ScenarioDocDecorator,
   ScenarioServiceDecorator,
-} from "../generated-defs/TypeSpec.SpecLib.js";
+} from "../../generated-defs/TypeSpec.SpecLib.js";
 import { SpecLibStateKeys } from "./lib.js";
```
Comment (sarangan12) [inline packages/spector/src/lib/decorators.ts:27]: Here alson from `SpecLibStateKeys` to `SpectorStateKeys`

**Pull request merged**